### DISCUSSION TITLE
Are both APIs (parse & gen) thread-safe?

### DISCUSSION THREAD

**Discussion by skunkiferous** [robohack/yajl#13] [General]
I haven't seen this covered in the doc, or the header files.

**Comment by robohack** [comment]
I guess it depends on exactly what you mean by "thread safe".

I think all the code should be fully reentrant.

I've used both the generator as well as `yajl_parse()` et al, simultaneoulsy, in some highly threaded applications, including regular test runs of some under Valgrind, all with no problem.

I had not used `yajl_tree_parse()` as much though (thus the more recent bug fixes therein).

**Comment by skunkiferous** [comment]
Thanks. I don't intend to use `yajl_tree_parse()` . I was looking for a C API to parse JSON, with user-defined number support (I need support for both int64 and uint64), "streaming" (can process input of unlimited size) and a permissive license. That basically only left YAJL, AFAIK.
